### PR TITLE
[monitoring-kubernetes] Added releases smart cache

### PR DESCRIPTION
**Pull request by Dmitrykob**
## Description

Added "in memory" cache to optimize the number of release secrets scans in the cluster.

## Why do we need it, and what problem does it solve?

The operation of scanning release secrets became duplicated when a new hook was added to control the ability to update the k8s version automatically.

## What is the expected result?

hooks can reuse secret scan results

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: monitoring-kubernetes
type: fix
summary: reducing the number of scans of release secrets in the cluster.
impact_level: default
```

### CODE REVIEW

File: modules/340-monitoring-kubernetes/hooks/helm.go [hunk 1]
```diff
@@ -122,9 +113,18 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
 	OnStartup: &go_hook.OrderedConfig{
 		Order: 1,
 	},
-}, dependency.WithExternalDependencies(handleHelmReleases))
+}, dependency.WithExternalDependencies(handleHelmReleasesByShedule))
+
+func handleHelmReleasesByShedule(input *go_hook.HookInput, dc dependency.Container) error {
```
Comment (yalosev) [inline modules/340-monitoring-kubernetes/hooks/helm.go:118]:
```suggestion
func handleHelmReleasesBySсhedule(input *go_hook.HookInput, dc dependency.Container) error {
```

File: modules/340-monitoring-kubernetes/hooks/helm.go [hunk 2]
```diff
@@ -122,9 +113,18 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
 	OnStartup: &go_hook.OrderedConfig{
 		Order: 1,
 	},
-}, dependency.WithExternalDependencies(handleHelmReleases))
+}, dependency.WithExternalDependencies(handleHelmReleasesByShedule))
```
Comment (yalosev) [inline modules/340-monitoring-kubernetes/hooks/helm.go:116]:
```suggestion
}, dependency.WithExternalDependencies(handleHelmReleasesBySсhedule))
```

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 8464060286 8464060286-1-con-1-25 /sys/deckhouse-oss/install:pr7558 cloud-user@51.250.9.114

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Pull request merged**